onedal/basic_statistics/basic_statistics.py

```diff
@@ -0,0 +1,99 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+from sklearn.base import BaseEstimator
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+from numbers import Number
+
+from ..common._policy import _get_policy
+
+from ..datatypes._data_conversion import (
+    from_table,
+    to_table,
+    _convert_to_supported,
+    _convert_to_dataframe)
+from onedal import _backend
+
+
+class BaseBasicStatistics(metaclass=ABCMeta):
+    @abstractmethod
+    def __init__(self, result_options, algorithm):
+        self.options = result_options
+        self.algorithm = algorithm
+
+    @staticmethod
+    def get_all_result_options():
+        return ["min", "max", "sum", "mean",
+                "variance", "variation", "sum_squares",
+                "standard_deviation", "sum_squares_centered",
+                "second_order_raw_moment"]
+
+    def _get_policy(self, queue, *data):
+        return _get_policy(queue, *data)
+
+    def _get_result_options(self, options):
+        if options == "all":
+            options = self.get_all_result_options()
+        if isinstance(options, list):
+            options = "|".join(options)
+        assert isinstance(options, str)
+        return options
+
+    def _get_onedal_params(self, dtype=np.float32):
+        options = self._get_result_options(self.options)
+        return {
+            'fptype': 'float' if dtype == np.float32 else 'double',
+            'method': self.algorithm, 'result_option': options,
+        }
+
+    def _compute(self, data, weights, module, queue):
+        policy = self._get_policy(queue, data, weights)
+
+        data_loc, weights_loc = _convert_to_dataframe(policy, data, weights)
+
+        data_loc, weights_loc = _convert_to_supported(
+            policy, data_loc, weights_loc)
```

Comment:
With this flow we probably firstly converting it to `numpy.float64` and then converting it to `numpy.float32` type ndarray. It is better do conversions at ones.

Reply:
I am still voting for reusing `_convert_to_supported`.

Reply:
I insist. It is part of functionality that will allow us to support different types of input. This part of library is crucial for now. I believe that even if we will find workaround and decide to delete this function - it still will be much faster & easier to identify such places by its name. Another thing is that it is clearly miscellaneous part of the library because it's name starts with an underscore.